app/scripts/lib/transaction/smart-transactions.test.ts

```diff
@@ -246,6 +246,22 @@ describe('submitSmartTransactionHook', () => {
     });
   });
 
+  it('falls back to regular transaction submit if it is a legacy transaction', async () => {
```

Comment:
This looks like a duplicated test (see the one above it).

I don't see this as coming from a recent v12.15.0 cherry-pick either, so I'm guessing it should be removed.